torch/fx/graph.py

```diff
@@ -648,8 +648,21 @@ def emit_node(node: Node):
                     "val",
                     node.meta.get("tensor_meta", node.meta.get("example_value", None)),
                 )
+                # DTensor support
+                if isinstance(meta_val, torch.distributed.tensor.DTensor):
```

Comment:
adding support for DTensor in tlparse is great, although:

(1)  I feel a bit bad about hardcoding DTensor into `torch.fx.Graph`

(2) it's a shame / might be confusing that we get the nice printing for a graph with DTensor inputs, but not e.g. `Float8Tensor` inputs

Reply:
one thing we could do is generaically check to see if the meta_val is a "traceable wrapper subclass", and if so, `__repr__` it. One potential risk of doing this though is that we are calling `repr()` on a subclass that may now hold FakeTensors / SymInts, which the subclass's `repr` may not handle well (what if their repr() calls `item()` on the inner tensor? they shouldn't do this, since it's not "PT2 friendly", but we probably shouldn't crash).

I think my take would be that we still do this generic `repr()` call for any pt2-traceable subclasses, but try/catch it and fall back to a slightly worse print if repr fails. How does that sound?

Reply:
that's a fair point, let me see if i can refactor this so all tensor subclasses can override a method to get nice tlparse printing